### PR TITLE
Close resources based on updated firebase_ml_vision

### PR DESCRIPTION
**Pull request by zura35**
Notes:
I found that detector would still be run after resources have been closed, hence I checked isDeactivated is true in _processImage().

### CODE REVIEW

**Review by jaumard — CHANGES_REQUESTED** [review]
If you want to close resources, do it on your own widget, you can't do it here because the plugin don't know which detector you use (and don't care).

With your PR you just broke the concept of the plugin because now it's hardcoded to use barcode detector where in fact it can use all firebase detector. .

It might be more useful to just add an `onDispose` callback on the plugin, and don't instantiate the barcode detector inside the plugin and put it as required instead. The callback will allow you to dispose your detector yourself. 

cc @Kleak

**Comment by zura35** [comment]
> If you want to close resources, do it on your own widget, you can't do it here because the plugin don't know which detector you use (and don't care).
> 
> With your PR you just broke the concept of the plugin because now it's hardcoded to use barcode detector where in fact it can use all firebase detector. .
> 
> It might be more useful to just add an `onDispose` callback on the plugin, and don't instantiate the barcode detector inside the plugin and put it as required instead. The callback will allow you to dispose your detector yourself.
> 
> cc @Kleak

In the plugin, if detector is not provided, the CameraMlVision state will create a BarcodeDetector as well on L74 in flutter_camera_ml_vision.dart. In the current version, that instance is not handled as well.

I agree that handling the detector should be done on one's own widget, and I would say if that's the case, it would be better if from the very beginning the default detector is never provided...?

**Comment by jaumard** [comment]
That's what I said :) 

> don't instantiate the barcode detector inside the plugin and put it as required instead

**Comment by zura35** [comment]
@jaumard I have updated the changes according to your suggestions, you may have a look, thanks!

File: lib/flutter_camera_ml_vision.dart [hunk 1]
```diff
@@ -216,6 +215,7 @@ class CameraMlVisionState<T> extends State<CameraMlVision<T>> {
 
   @override
   void dispose() {
+    if (widget.onDispose != null) widget.onDispose();
```
Comment (jaumard) [inline lib/flutter_camera_ml_vision.dart]: not sure what the analyzer say about this but we generally put bracket even for one line generally

**Review by jaumard — DISMISSED** [review]
LGTM :)

**Comment by jaumard** [comment]
I prefer like this :) thanks for the PR @zura35 !

be careful @Kleak when you publish, this is a breaking change

File: lib/flutter_camera_ml_vision.dart (hunk 1 above)
Comment (zura35) [reply lib/flutter_camera_ml_vision.dart]: As resolved in https://github.com/rushio-consulting/flutter_camera_ml_vision/pull/67/commits/a3e924a22edea44b07a84ee1005b0aa07bc4fa49

**Comment by kevin-sakemaer** [comment]
@zura35 Hi could you fix the merge request we will merge it and publish it right after as 3.0.0

**Review by jaumard — APPROVED** [review]
LGTM

**Pull request merged**